### ISSUE TITLE
Intermittent NullPointerException: link-destination

### ISSUE THREAD

**Issue by damianw** [noties/Markwon#171]
* **Markwon version**: 4.1.1

We are occasionally seeing reports of the following error during rendering. It's very intermittent, and is probably some kind of race condition (we are rendering multiple documents in parallel using the same `Markwon` instance [as previously mentioned in this other issue](https://github.com/noties/Markwon/issues/157)). I don't think I can really provide more information as I haven't yet been able to reproduce it manually.

```
Caused by java.lang.NullPointerException: link-destination
       at io.noties.markwon.Prop.require(Prop.java:52)
       at io.noties.markwon.core.factory.LinkSpanFactory.getSpans(LinkSpanFactory.java:18)
       at io.noties.markwon.MarkwonVisitorImpl.setSpansForNodeOptional(MarkwonVisitorImpl.java:261)
       at io.noties.markwon.MarkwonVisitorImpl.setSpansForNodeOptional(MarkwonVisitorImpl.java:254)
       at io.noties.markwon.core.CorePlugin$15.visit(CorePlugin.java:514)
       at io.noties.markwon.core.CorePlugin$15.visit(CorePlugin.java:502)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:172)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:125)
       at org.commonmark.node.Link.accept(Link.java:38)
       at io.noties.markwon.MarkwonVisitorImpl.visitChildren(MarkwonVisitorImpl.java:203)
       at io.noties.markwon.core.CorePlugin$14.visit(CorePlugin.java:474)
       at io.noties.markwon.core.CorePlugin$14.visit(CorePlugin.java:463)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:172)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:140)
       at org.commonmark.node.Paragraph.accept(Paragraph.java:7)
       at io.noties.markwon.MarkwonVisitorImpl.visitChildren(MarkwonVisitorImpl.java:203)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:174)
       at io.noties.markwon.MarkwonVisitorImpl.visit(MarkwonVisitorImpl.java:75)
       at org.commonmark.node.Document.accept(Document.java:7)
       at io.noties.markwon.MarkwonImpl.render(MarkwonImpl.java:66)
```

**Comment by noties** [comment]
Hello @damianw !

Thank you for the hint. `RenderProps` were still re-used between multiple `render` calls. I've fixed that and published `4.1.2-SNAPSHOT` version

**Comment by noties** [comment]
Hello @damianw !

As you can imagine this kind of issue is hard to track. I could not reproduce it with code, but there is a big chance that this behavior was caused by `RenderProps` being shared between multiple visitors. This was an oversight and I think this should be fixed in `4.1.2` release that I had just published 🙌

**Comment by damianw** [comment]
Thanks @noties! I will update and see if it fixes the issue. 👍
